nomenclature/region_mapping_models.py

```diff
@@ -165,19 +181,39 @@ def __init__(
         super().__init__(self.template.format(model=mapping1.model, files=files))
 
 
+class RegionNotDefinedError(ValueError):
+    template = "Region(s) {region} in {file} not defined"
+
+    def __init__(self, region, file) -> None:
+        self.file = file
+        super().__init__(self.template.format(region=region, file=file))
+
+
 class RegionProcessor(BaseModel):
     """Region aggregation mappings for scenario processing"""
 
+    dsd: DataStructureDefinition
     mappings: Dict[str, RegionAggregationMapping]
 
+    class Config:
+        # necessary since DataStructureDefinition is not a pydantic class
+        arbitrary_types_allowed = True
+
     @classmethod
-    @validate_arguments
-    def from_directory(cls, path: DirectoryPath):
+    @validate_arguments(config=dict(arbitrary_types_allowed=True))
```

Comment:
Just asking - do you really need both the extra class and the config-dict?

Reply:
Yes because I am asking pydantic to validate the function inputs and one of the inputs is `DataStructureDefinition` which is not a pydantic class. Therefore pydantic would throw an error as it attempts to validate it so I need to set `arbitrary_types_allowed` to `True`. See [pydantic documentation](https://pydantic-docs.helpmanual.io/usage/types/#arbitrary-types-allowed) and this [stackoverflow post](https://stackoverflow.com/questions/68349150/best-practice-to-pydantic-validate-arguments-for-non-built-in-types-like-pandas) about the `@validate_arguments` decorator specifically.